packages/ragbits-core/src/ragbits/core/prompt/prompt.py

```diff
@@ -181,6 +183,44 @@ def list_few_shots(self) -> ChatFormat:
             result.append({"role": "assistant", "content": assistant_message})
         return result
 
+    def list_images(self) -> list[dict]:
+        """
+        Returns the schema of the list of images compatible with llm apis
+        Returns:
+            list of dictionaries
+        """
+        image_prompt_list = (
+            [
+                {
+                    "type": "image_url",
+                    "image_url": {"url": f"data:image/png;base64 {base64.b64encode(image).decode('utf-8')}"},
+                }
+                for image in self.images
+            ]
+            if self.images
+            else []
+        )
+        return image_prompt_list
+
+    def get_user_message(self) -> ChatFormat:
+        """
+        Parses the rendered user prompt and combines it with images if they exist
+        Returns:
+            ChatFormat user_message
+        """
+        images = self.list_images()
+        if not images:
+            user_message = {
+                "role": "user",
+                "content": self.rendered_user_prompt,
+            }
+        else:
+            user_message = {
+                "role": "user",
+                "content": [{"type": "text", "text": self.rendered_user_prompt}, *images],  # type: ignore
+            }  # type: ignore
```

Comment:
Why is `# type: ignore` necessary here?

Reply:
this has been removed from here however `# type: ignore` is used at the other part of the code - the reason is that dict representing image has different struture than representing text - mypy keeps throwing errors if not used.